### PR TITLE
[ServerManage] Cog model restructure

### LINKED ISSUE #619: [ServerManage] Cog model restructure

### PR DESCRIPTION
**Pull request by quachtridat**
## Summary ##

This PR resolves #619 by restructuring cog `ServerManage` according to SFUAnime/Ren#608. No function changes are made in this PR.

## Details ##

Restructure cog `ServerManage` according to SFUAnime/Ren#608 by:
- Moving all methods from class `ServerManage` to `Core`
- Decoupling core command handling method calls from Red's command handler decorators by splitting `commands.py` into `commandsCore.py` and `commandHandlers.py`
- Dropping `ServerManageMeta` metaclass

Prior to this PR, `ServerManage`'s cog class (`commands.Cog`-derived class) inherited `ServerManageCommandHandlers` - a command handling class whose instance methods made calls to methods that existed in the `ServerManage` class. This worked at runtime, but static type checkers would fail to resolve definitions for `ServerManage` methods called within `ServerManageCommandHandlers`. Following the updated cog model structure (#608), this will no longer be a problem.

## Manual testing ##

This is done with an assumption that the core functions are working well, which is true as long as there has been no problem with any of them prior to this PR.

### Load + help ###

<details>
<summary>Images</summary>

image
image
image
image

</details>

### Banners ###

<details>
<summary>Images</summary>

image
image
image
image
image

</details>

### Icons ###

<details>
<summary>Images</summary>

image
image
image

</details>

#### Before the assigned date ####

<details>
<summary>Images</summary>

image
image

</details>

#### After the assigned date ####

No banner change 'cause my testing server has no boost :sadge:

<details>
<summary>Images</summary>

image

</details>

<details>
<summary>Log lines</summary>

```
[00:01:32] INFO     [red.luicogs.ServerManage] Checking to see if we need to change server images
           DEBUG    [red.luicogs.ServerManage] Checking guild icon for [Server Name] ([Server ID])
           INFO     [red.luicogs.ServerManage] Changed the server icon for [Server Name] ([Server ID]) to test
           DEBUG    [red.luicogs.ServerManage] Checking guild banner for [Server Name] ([Server ID])
[00:01:33] INFO     [red.luicogs.ServerManage] Changed the server banner for [Server Name] ([Server ID]) to test
```

</details>

### Unload ###

<details>
<summary>Images</summary>

image

</details>

### Logs ###

`[Discord Username]`, `[Discord User ID]`, `[Server Name]`, and `[Server ID]` are just redacted information.

<details>
<summary>Log lines</summary>

```
[22:57:45] INFO     [red.luicogs.ServerManage] Checking to see if we need to change server images
           DEBUG    [red.luicogs.ServerManage] Checking guild icon for [Server Name] ([Server ID])
           DEBUG    [red.luicogs.ServerManage] Checking guild banner for [Server Name] ([Server ID])
[23:06:43] INFO     [red.luicogs.ServerManage] Checking to see if we need to change server images
           DEBUG    [red.luicogs.ServerManage] Checking guild icon for [Server Name] ([Server ID])
           DEBUG    [red.luicogs.ServerManage] Checking guild banner for [Server Name] ([Server ID])
[23:06:45] ERROR    [red.luicogs.ServerManage] We should never reach this!
[23:07:06] INFO     [red.luicogs.ServerManage] User [Discord Username] ([Discord User ID]) added a(n) banner 'test.png'
[23:07:59] INFO     [red.luicogs.ServerManage] Checking to see if we need to change server images
           DEBUG    [red.luicogs.ServerManage] Checking guild icon for [Server Name] ([Server ID])
           DEBUG    [red.luicogs.ServerManage] Checking guild banner for [Server Name] ([Server ID])
[23:08:43] ERROR    [red.luicogs.ServerManage] We should never reach this!
[23:09:47] INFO     [red.luicogs.ServerManage] User [Discord Username] ([Discord User ID]) deleted a(n) banner 'test.png'
[23:11:04] INFO     [red.luicogs.ServerManage] User [Discord Username] ([Discord User ID]) added a(n) icon 'test.png'
[23:11:35] INFO     [red.luicogs.ServerManage] User [Discord Username] ([Discord User ID]) deleted a(n) icon 'test.png'
[23:20:00] ERROR    [red.luicogs.ServerManage] We should never reach this!
```

</details>

The error log line

> `ERROR    [red.luicogs.ServerManage] We should never reach this!`

appeared every time I reloaded/unloaded the cog. There were no breakages observed during manual testing.

### CODE REVIEW

File: cogs/servermanage/core.py [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+        self.config.register_guild(**BASE_GUILD)
+
+        # Initialize logger, and save to cog folder.
+        self.dataFolder = data_manager.cog_data_path(cog_instance=self)
+        self.logger = logging.getLogger("red.luicogs.ServerManage")
+        if not self.logger.handlers:
+            logPath = pathJoin(self.dataFolder, "info.log")
+            handler = logging.FileHandler(filename=logPath, encoding="utf-8", mode="a")
+            handler.setFormatter(
+                logging.Formatter("%(asctime)s %(message)s", datefmt="[%d/%m/%Y %H:%M:%S]")
```
Comment (Injabie3) [inline cogs/servermanage/core.py]: Not directly part of this PR, but can you fix this to be YYYY-MM-DD?

**Review by Injabie3 — COMMENTED** [review]
LGTM, can you please do a sanity check of server icon change only to ensure this part still works?

File: cogs/servermanage/core.py (hunk 1 above)
Comment (quachtridat) [reply cogs/servermanage/core.py]: Sure.

**Comment by quachtridat** [comment]
> can you please do a sanity check of server icon change only to ensure this part still works?

Added this to the manual testing section:

### Banner and icon change ###

#### Before the assigned date ####

<details>
<summary>Images</summary>

image
image

</details>

#### After the assigned date ####

No banner change 'cause my testing server has no boost :sadge:

<details>
<summary>Images</summary>

image

</details>

<details>
<summary>Log lines</summary>

```
[00:01:32] INFO     [red.luicogs.ServerManage] Checking to see if we need to change server images
           DEBUG    [red.luicogs.ServerManage] Checking guild icon for [Server Name] ([Server ID])
           INFO     [red.luicogs.ServerManage] Changed the server icon for [Server Name] ([Server ID]) to test
           DEBUG    [red.luicogs.ServerManage] Checking guild banner for [Server Name] ([Server ID])
[00:01:33] INFO     [red.luicogs.ServerManage] Changed the server banner for [Server Name] ([Server ID]) to test
```

</details>

**Review by Injabie3 — APPROVED** [review]

**Pull request merged**